spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -377,12 +378,23 @@ protected MessageHandler createProducerMessageHandler(
 		if (StringUtils.hasText(producerProperties.getExtension().getRecordMetadataChannel())) {
 			handler.setSendSuccessChannelName(producerProperties.getExtension().getRecordMetadataChannel());
 		}
+
 		KafkaHeaderMapper mapper = null;
 		if (this.configurationProperties.getHeaderMapperBeanName() != null) {
 			mapper = getApplicationContext().getBean(
 					this.configurationProperties.getHeaderMapperBeanName(),
 					KafkaHeaderMapper.class);
 		}
+		if (mapper == null) {
+			//First, try to see if there is a bean named headerMapper registered by other frameworks using the binder (for e.g. spring cloud sleuth)
+			try {
+				mapper = getApplicationContext().getBean("kafkaHeaderMapper", KafkaHeaderMapper.class);
```

Comment:
I think this name should have something to do with the project. `kafkaBinderHeaderMapper` ?

Reply:
Good idea. Will update.